crates/iceberg/src/arrow/schema.rs

```diff
@@ -378,7 +378,24 @@ impl ArrowSchemaVisitor for ArrowSchemaConverter {
             DataType::Int8 | DataType::Int16 | DataType::Int32 => {
                 Ok(Type::Primitive(PrimitiveType::Int))
             }
+            // Cast unsigned types based on bit width (following Python implementation)
```

Comment:
```suggestion
            // Cast unsigned types based on bit width to allow for no data loss
```
I'm sure python compatibility is a direct goal here?

Reply:
Will incorporate this